src/model/HeatPumpPlantLoopEIRCooling.cpp

```diff
@@ -0,0 +1,626 @@
+    }
+
+    boost::optional<Node> HeatPumpPlantLoopEIRCooling_Impl::sourceSideWaterInletNode() const {
+      boost::optional<Node> result;
+
+      if (auto mo = demandInletModelObject()) {
+        result = mo->optionalCast<Node>();
+      }
+      return result;
+    }
+
+    boost::optional<Node> HeatPumpPlantLoopEIRCooling_Impl::sourceSideWaterOutletNode() const {
+      boost::optional<Node> result;
+
+      if (auto mo = demandOutletModelObject()) {
+        result = mo->optionalCast<Node>();
+      }
+      return result;
+    }
+
+    boost::optional<Node> HeatPumpPlantLoopEIRCooling_Impl::loadSideWaterInletNode() const {
+      boost::optional<Node> result;
+
+      if (auto mo = supplyInletModelObject()) {
+        result = mo->optionalCast<Node>();
+      }
+      return result;
+    }
+
+    boost::optional<Node> HeatPumpPlantLoopEIRCooling_Impl::loadSideWaterOutletNode() const {
+      boost::optional<Node> result;
+
+      if (auto mo = supplyOutletModelObject()) {
+        result = mo->optionalCast<Node>();
+      }
+      return result;
+    }
+
+    bool HeatPumpPlantLoopEIRCooling_Impl::addToNode(Node& node) {
+
+      // call the base class implementation to connect the component
+      bool ok = WaterToWaterComponent_Impl::addToNode(node);
+
+      // If there's a secondary plant loop, switch the condenser type to "WaterCooled"
+      if (this->sourceSideWaterLoop()) {
+        this->setCondenserType("WaterSource");
+      }
+      return ok;
+    }
+
+    bool HeatPumpPlantLoopEIRCooling_Impl::removeFromSecondaryPlantLoop() {
+      // Disconnect the component
+      bool ok = WaterToWaterComponent_Impl::removeFromSecondaryPlantLoop();
+
+      // Switch the condenser type back to "AirSource"
+      this->setCondenserType("AirSource");
+      return ok;
+    }
+
+    std::string HeatPumpPlantLoopEIRCooling_Impl::condenserType() const {
+      boost::optional<std::string> value = getString(OS_HeatPump_PlantLoop_EIR_CoolingFields::CondenserType, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    bool HeatPumpPlantLoopEIRCooling_Impl::setCondenserType(const std::string& condenserType) {
+      bool ok = false;
+      if (openstudio::istringEqual("AirSource", condenserType) && (this->secondaryPlantLoop())) {
+        LOG(Warn, "Cannot set condenserType to AirSource, HeatPumpPlantLoopEIRCooling '"
+                    << this->name() << "' is connected to a Source Side Loop. Use removeFromSecondaryPlantLoop() instead.");
+      } else if (istringEqual("WaterSource", condenserType) && !(this->secondaryPlantLoop())) {
+        LOG(Warn, "Cannot set condenserType to 'WaterSource', HeatPumpPlantLoopEIRCooling '"
+                    << this->name() << "' is not connected to a Source Side Loop. Use addToNode(PlantLoop&) instead.");
+      } else {
+        ok = setString(OS_HeatPump_PlantLoop_EIR_CoolingFields::CondenserType, condenserType);
+      }
+
+      return ok;
+    }
```

Comment:
Enforce condenserType in `addToNode` / `removeFromSecondaryPlantLoop` / `setCondenserType` (you should never call setCondenserType for this object, since there are only two options, unlike chillerElectricEIR)